src/main/java/com/mraof/minestuck/network/computer/BurnDiskPacket.java

```diff
@@ -0,0 +1,78 @@
+package com.mraof.minestuck.network.computer;
+
+import com.mraof.minestuck.blockentity.ComputerBlockEntity;
+import com.mraof.minestuck.computer.ProgramData;
+import com.mraof.minestuck.network.PlayToServerPacket;
+import net.minecraft.core.BlockPos;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.entity.item.ItemEntity;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.state.BlockState;
+
+import java.util.Random;
+
+public class BurnDiskPacket implements PlayToServerPacket
+{
+	private final BlockPos bePos;
+	private final int programId;
+	
+	public BurnDiskPacket(BlockPos pos, int programId)
+	{
+		this.bePos = pos;
+		this.programId = programId;
+	}
+	
+	public static BurnDiskPacket create(ComputerBlockEntity be, int programId)
+	{
+		return new BurnDiskPacket(be.getBlockPos(), programId);
+	}
+	
+	@Override
+	public void encode(FriendlyByteBuf buffer)
+	{
+		buffer.writeBlockPos(bePos);
+		buffer.writeInt(programId);
+	}
+	
+	public static BurnDiskPacket decode(FriendlyByteBuf buffer)
+	{
+		BlockPos bePos = buffer.readBlockPos();
+		int programId = buffer.readInt();
+		return new BurnDiskPacket(bePos, programId);
+	}
+	
+	@Override
+	public void execute(ServerPlayer player)
+	{
+		Level level = player.level;
+		if(level.isAreaLoaded(bePos, 0))    //TODO also check distance to the computer pos (together with a continual check clientside)
```

Comment:
I'd suggest using `ComputerBlockEntity.forNetworkIfPresent()` here

Reply:
Im not sure how to handle the creation of the disk item and other context specific to the disk burning using that function

Reply:
The function takes a block entity consumer as an argument. You'd use the computer block entity inside this consumer as you're right now using the block entity after the instanceof check.

Reply:
Should be good now but will not resolve until you have approved how its been changed